Ladybird+Browser: Move config files from /home/anon/.config to /res/ladybird

bplaat opened the pull request:
This is a follow up of #26164, currently new users coudn't start the Browser because of missing config files. I've moved these files to `/res/ladybird`, and updated that folder layout as it currently exists in the Ladybird repo (https://github.com/LadybirdBrowser/ladybird/tree/master/Base/res/ladybird). This results in that a new user can open the Browser 🎉.

I also changed that by default the Browser bookmarkers bar is hidden because it's empty for new users. Except for anon because there is already a filled `bookmarkers.json` file.

These changes also work fine in Lagom.

bplaat committed:
Ladybird+LibWeb: Move about pages from anon config to /res

nico commented:
Haven't looked at the diff, but let's use /res/browser. Ladybird is the standalone project.

bplaat committed:
Ladybird+Browser: Move default config files to /res

bplaat committed:
Browser: Hide bookmarkers bar by default because it's empty

Except for the anon user that already has a prefilled
bookmarkers.json file in the .config folder.

bplaat committed:
Ladybird+Browser: Move browser resources to /res/browser

trflynn89 commented:
Isn't `/res` read-only? How does writing to these config files work?

bplaat commented:
> Isn't /res read-only? How does writing to these config files work?

You are totally right, I have changed the code to copy the both default config files to `.config/` on first launch

bplaat commented:
This also fixes #26604
